src/borrow/mod.rs

```diff
@@ -1,4 +1,33 @@
-//! Types and traits for obtaining temporary access to the internals of JavaScript values.
+//! Provides temporary access to JavaScript typed arrays.
+//!
+//! ## Typed Arrays
+//!
+//! JavaScript's [typed arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Typed_arrays)
+//! are objects that allow reading and writing raw binary data in memory.
+//!
+//! Typed arrays are managed with the
+//! [`ArrayBuffer`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer)
+//! type, which controls the storage of the underlying data buffer,
+//! and several typed views for managing access to the buffer.
+//! Neon provides access to the `ArrayBuffer` class with the
+//! [`JsArrayBuffer`](../../neon/types/struct.JsArrayBuffer.html) type.
+//!
+//! Node also provides a [`Buffer`](https://nodejs.org/api/buffer.html) type,
+//! which is built on top of `ArrayBuffer` and provides additional functionality.
+//! Neon provides access to the `Buffer` class with the
+//! [`JsBuffer`](../../neon/types/struct.JsBuffer.html) type.
+//!
+//! Many of Node's I/O APIs work with these types, and they can also be used for
```

Comment:
It may be beneficial to mention that they can be used to efficiently share data between JavaScript and Rust _without_ copying.